Sources/Core/AWSClientRuntime/Middlewares/Sha256TreeHashMiddleware.swift

```diff
@@ -37,16 +37,14 @@ public struct Sha256TreeHashMiddleware<OperationStackOutput: HttpResponseBinding
                   let streamBytes: Data?
                   let currentPosition = stream.position
                   if stream.isSeekable {
-                      streamBytes = try stream.readToEnd()
+                      streamBytes = try await stream.readToEndAsync()
                       try stream.seek(toOffset: currentPosition)
                   } else {
                       // If the stream is not seekable, we need to cache the stream in memory
                       // so we can compute the hash and still be able to send the stream to the service.
                       // This is not ideal, but it is the best we can do.
-                      let cachingStream = CachingStream(base: stream)
-                      streamBytes = try cachingStream.readToEnd()
-                      try cachingStream.seek(toOffset: currentPosition)
-                      input.withBody(.stream(cachingStream))
+                      streamBytes = try await stream.readToEndAsync()
+                      input.withBody(.data(streamBytes))
```

Comment:
This is another place where `CachingStream` was used but is unnecessary.  The stream body is just replaced with the data that was read from the stream.